tests/integration_tests/test_docker_executable.py

```diff
@@ -0,0 +1,297 @@
+
+"""Integration tests which leverage the source-faker connector to test the framework end-to-end with the Docker Executor."""
+
+from __future__ import annotations
+
+import os
+import shutil
+import tempfile
+from collections.abc import Generator
+from pathlib import Path
+
+import airbyte as ab
+import pytest
+import ulid
+from airbyte.caches.base import CacheBase
+from airbyte.caches.duckdb import DuckDBCache
+from airbyte.caches.postgres import PostgresCache
+from airbyte.caches.util import new_local_cache
+
+# Product count is always the same, regardless of faker scale.
+NUM_PRODUCTS = 100
+
+SEED_A = 1234
+SEED_B = 5678
+
+# Number of records in each of the 'users' and 'purchases' streams.
+FAKER_SCALE_A = 200
+# We want this to be different from FAKER_SCALE_A.
+FAKER_SCALE_B = 300
+
+
+# Patch PATH to include the source-faker executable.
+
+
+@pytest.fixture(scope="function")  # Each test gets a fresh source-faker instance.
+def source_faker_seed_a() -> ab.Source:
+    """Fixture to return a source-faker connector instance."""
+    source = ab.get_source(
+        "source-faker",
+        docker_executable=True,
+        config={
+            "count": FAKER_SCALE_A,
+            "seed": SEED_A,
+            "parallelism": 16,  # Otherwise defaults to 4.
+        },
+    )
+    source.check()
+    source.select_streams([
+        "users",
+        "products",
+        "purchases",
+    ])
+    return source
+
+
+@pytest.fixture(scope="function")  # Each test gets a fresh source-faker instance.
+def source_faker_seed_b() -> ab.Source:
+    """Fixture to return a source-faker connector instance."""
+    source = ab.get_source(
+        "source-faker",
+        docker_executable=True,
+        config={
+            "count": FAKER_SCALE_B,
+            "seed": SEED_B,
+            "parallelism": 16,  # Otherwise defaults to 4.
+        },
+    )
+    source.check()
+    source.select_streams([
+        "products",
+        "purchases",
+        "users",
+    ])
+    return source
+
+
+@pytest.fixture(scope="function")
+def duckdb_cache() -> Generator[DuckDBCache, None, None]:
+    """Fixture to return a fresh cache."""
+    cache: DuckDBCache = new_local_cache()
+    yield cache
+    # TODO: Delete cache DB file after test is complete.
+    return
+
+
+@pytest.fixture(scope="function")
+def postgres_cache(new_postgres_cache) -> Generator[PostgresCache, None, None]:
+    """Fixture to return a fresh cache."""
+    yield new_postgres_cache
+    # TODO: Delete cache DB file after test is complete.
+    return
+
+
+@pytest.fixture
+def all_cache_types(
+    duckdb_cache: DuckDBCache,
+    postgres_cache: PostgresCache,
+):
+    _ = postgres_cache
+    return [
+        duckdb_cache,
+        postgres_cache,
+    ]
+
+
+def test_which_source_faker() -> None:
+    """Test that source-faker is available on PATH."""
+    assert shutil.which(
+        "source-faker"
+    ), f"Can't find source-faker on PATH: {os.environ['PATH']}"
+
+
```

Comment:
We can delete this test since it doesn't actually test the docker invocation path (happens to pass but for other reasons).

```suggestion
```